airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/OAuthHandler.java

```diff
@@ -361,4 +361,51 @@ JsonNode getOAuthInputConfiguration(final JsonNode hydratedSourceConnectionConfi
     return Jsons.jsonNode(result);
   }
 
+  /**
+   * Given an OAuth response, writes a secret and returns the secret Coordinate in the appropriate format.
+   * <p>
+   * Unlike our regular source creation flow, the OAuth credentials created and stored this way will be stored
+   * in a singular secret as a string. When these secrets are used, the user will be expected to use the
+   * specification to rehydrate the connection configuration with the secret values prior to saving a
+   * source/destination.
+   * <p>
+   * The singular secret was chosen to optimize UX for public API consumers (passing them one secret
+   * to keep track of > passing them a set of secrets).
+   * <p>
+   * See https://github.com/airbytehq/airbyte/pull/22151#discussion_r1104856648 for full discussion.
+   */
+  public Map<String, Object> writeOAuthResponseSecret(final UUID workspaceId, final Map<String, Object> payload) {
+
+    try {
+      final String payloadString = Jackson.getObjectMapper().writeValueAsString(payload);
+      final SecretCoordinate secretCoordinate = secretsRepositoryWriter.storeSecret(generateOAuthSecretCoordinate(workspaceId), payloadString);
+      return formatOAuthSecretResponse(secretCoordinate);
+    } catch (final JsonProcessingException e) {
+      throw new RuntimeException("Json object could not be written to string.", e);
+    }
+  }
+
+  public Map<String, Object> formatOAuthSecretResponse(final SecretCoordinate secretCoordinate){
```

Comment:
- this can be a `private` function since it's not used outside the class.
- also can probably fold this into the calling function as it's one line for now.

Reply:
This makes sense, folded it back into the calling function.